### PR TITLE
154: Lesson 4 Page

### LINKED ISSUE #154: Feature: Make Lesson 4 Page- Dereferencing Pointers
Make the lesson 4 page based on the Figma:
Screenshots here-
image
image
image
Note that the center image with the partially filled box flowing into the other box should be an animation, but that will come later- so either include a static image for now or just leave a comment of where the component should go.

### PR DESCRIPTION
**Pull request by ajtadeo**
## Summary

- add Lesson 4 content and style
- add Lesson 4 to SectionHeader enum, app router, and PageOrder enum
- organized PageWrapper.scss and added new rules
- make Terminal text smaller (21px)

image

image

Closes #154

### CODE REVIEW

**Comment by ajtadeo** [comment]
@Dankco please merge the Note component #170 first so I can implement it in Lesson 4 page. Thanks!

**Comment by Dankco** [comment]
Merged.

**Comment by ajtadeo** [comment]
Updated lesson page with Note component, ready for review!

image

File: src/pages/Lesson4.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+          </div>
+          <p>
+            Now you might be wondering, what good will knowing an address do?
+            And you&apos;re right - most of the time, we don&apos;t care about
+            the actual value of the address itself, we care about the value
+            stored at that address. So if we are just given an address, it would
+            be really useful if there were a way for us to find out what value
+            is stored there. And you guessed it, there is!
+          </p>
+          <label>Boxed Syntax Notation:</label>
```
Comment (Dankco) [inline src/pages/Lesson4.tsx]: Sorry, should have caught this earlier on the Figma, Lesson 2 page 2, and Lesson 3, but this label was just for indicating the design, not for inclusion, Can you remove this from this lesson as well as from Lesson 3? I'll write a Note up to quickly explain the boxed syntax.

File: src/styles/PageWrapper.scss [hunk 2]
```diff
@@ -19,31 +19,65 @@ $layout-hspacing: 13vh;
...
+
+  h1 + .diagram {
     padding: 4vh 0;
   }
 
+  .diagram + h1 {
+    padding: 0;
+  }
+
+  // label
```
Comment (Dankco) [inline src/styles/PageWrapper.scss]: rm as per above

**Comment by ajtadeo** [comment]
Removed "Boxed Syntax Notation" label for lesson 2.2 and lesson 4

image

image

**Review by Dankco — APPROVED** [review]
image
I'll ticket this since it seems like the sentence was not completed here.

**Pull request merged**